ray-operator/controllers/ray/common/route.go

```diff
@@ -11,74 +14,157 @@ import (
-	for key, value := range cluster.Annotations {
-		annotation[key] = value
-	}
-
-	servicePorts := getServicePorts(cluster)
-	dashboardPort := utils.DefaultDashboardPort
-	if port, ok := servicePorts["dashboard"]; ok {
-		dashboardPort = int(port)
-	}
-
 	weight := int32(100)
 
 	serviceName, err := utils.GenerateHeadServiceName("RayCluster", cluster.Spec, cluster.Name)
 	if err != nil {
 		return nil, err
 	}
 
-	route := &routev1.Route{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:        utils.GenerateRouteName(cluster.Name),
-			Namespace:   cluster.Namespace,
-			Labels:      labels,
-			Annotations: annotation,
-		},
-		Spec: routev1.RouteSpec{
-			To: routev1.RouteTargetReference{
-				Kind:   "Service",
-				Name:   serviceName,
-				Weight: &weight,
+	ingressOptionsList := cluster.Spec.HeadGroupSpec.IngressOptions.Ingresses
+	if len(ingressOptionsList) == 0 {
+		// Copy other configurations from cluster annotations to provide a generic way
+		// for user to customize their route settings.
+		annotation := map[string]string{}
+		for key, value := range cluster.Annotations {
+			annotation[key] = value
+		}
+
+		servicePorts := getServicePorts(cluster)
+		dashboardPort := utils.DefaultDashboardPort
+		if port, ok := servicePorts["dashboard"]; ok {
+			dashboardPort = int(port)
+		}
+
+		route := &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        utils.GenerateRouteName(cluster.Name),
+				Namespace:   cluster.Namespace,
+				Labels:      labels,
+				Annotations: annotation,
 			},
-			Port: &routev1.RoutePort{
-				TargetPort: intstr.FromInt(dashboardPort),
+			Spec: routev1.RouteSpec{
+				To: routev1.RouteTargetReference{
+					Kind:   "Service",
+					Name:   serviceName,
+					Weight: &weight,
+				},
+				Port: &routev1.RoutePort{
+					TargetPort: intstr.FromInt(dashboardPort),
+				},
+				WildcardPolicy: "None",
 			},
-			WildcardPolicy: "None",
-		},
-	}
+		}
+		routeList = append(routeList, route)
+		return routeList, nil
+	} else {
+		for _, ingressItem := range ingressOptionsList {
+			annotations := ingressItem.Annotations
+			tlsConfigs := ingressItem.TLSConfig
+			port := ingressItem.Port
+			annotation := map[string]string{}
+			tlsConfig := map[string]string{}
+
+			for key, value := range annotations {
+				annotation[key] = value
+			}
+
+			for key, value := range tlsConfigs {
+				tlsConfig[key] = value
+			}
 
-	return route, nil
+			route := &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        ingressItem.IngressName,
+					Annotations: annotation,
+					Namespace:   cluster.Namespace,
+					Labels:      labels,
+				},
+				Spec: routev1.RouteSpec{
+					To: routev1.RouteTargetReference{
+						Kind:   "Service",
+						Name:   serviceName,
+						Weight: &weight,
+					},
+					Port: &routev1.RoutePort{
+						TargetPort: intstr.FromInt(port),
+					},
+					WildcardPolicy: "None",
+				},
+			}
+			if len(annotation) > 0 {
+				route.ObjectMeta.Annotations = annotation
+			}
```

Comment:
I think this can be removed, as the annotation for the route is already set anyways above.